### ISSUE TITLE
[Feat] r.surf.* need seed parameter for reproducibility

### ISSUE THREAD

**Issue by petrasovaa** [OSGeo/grass#1044]
Modules r.surf.random, r.surf.gauss, r.surf.fractal generate different results every time you run them.
They need seed parameter for reproducible results, similarly to r.mapcalc.

**Comment by istutea** [comment]
Hi @petrasovaa i'm looking forward for working on Adding JSON output project as part of GSoC ,can i start working on this issue ?

**Comment by petrasovaa** [comment]
> Hi @petrasovaa i'm looking forward for working on Adding JSON output project as part of GSoC ,can i work on this issue ?

Welcome! Yes, I suggest to work on r.surf.fractal, look at how r.mapcalc or r.random deal with it using seed parameter and -s flag. Ideally please also include a test. Let us know if you have questions.

**Comment by Shashankss1205** [comment]
Hi @petrasovaa , I am Shashank Shekhar Singh a sophomore from IIT BHU, India. I have developed interest in this project and want to contribute to `Add JSON output to different tools in C` as my GSOC Project this year. Hope we can discuss further procedures for the same.

**Comment by Shashankss1205** [comment]
> > Hi @petrasovaa i'm looking forward for working on Adding JSON output project as part of GSoC ,can i work on this issue ?
> 
> Welcome! Yes, I suggest to work on r.surf.fractal, look at how r.mapcalc or r.random deal with it using seed parameter and -s flag. Ideally please also include a test. Let us know if you have questions.

Hi @petrasovaa, I have written the code for adding seed and -s flag for reproducibility, along with a unit test (please verify it once).
I tried sending a PR for the same, but it says Permission denied.
![Screenshot from 2024-03-10 03-22-06](https://github.com/OSGeo/grass/assets/123410790/54e6ec79-6fcb-41fe-93f2-eb79155728d1)


Can you please help me out !!

**Comment by echoix** [comment]
You are trying to push directly to the OSGeo/grass repo, not your fork.
Try using the little cloud with an up arrow in the bottom left corner. It should/might offer you to push that branch in your fork

**Comment by Shashankss1205** [comment]
Oh I am extremely sorry for doing this silly mistake here.
Sending a forked PR in a moment.

**Comment by kritibirda26** [comment]
Hi @petrasovaa I intend to apply to "Adding JSON output project as part of GSoC" as well but I see a PR is already open for this issue. Should I work on this and open a PR for it anyway? Otherwise, can you please point me to another issue to work on? Thanks.

**Comment by petrasovaa** [comment]
> Hi @petrasovaa I intend to apply to "Adding JSON output project as part of GSoC" as well but I see a PR is already open for this issue. Should I work on this and open a PR for it anyway? Otherwise, can you please point me to another issue to work on? Thanks.

Sure. I would suggest to start working on adding JSON to one of the tools mentioned in #3020. Specifically, I suggest adding JSON to r.what. I created an issue for this: #3518, so we can discuss more there.

**Comment by Jasmitha-Mannam** [comment]
Hi...... , I want to work on this problem problem statement as a part of GSOC

**Comment by petrasovaa** [comment]
> Hi...... , I want to work on this problem problem statement as a part of GSOC

Hi @Jasmitha-Mannam, we currently don't have anybody else interested in the [STAC idea](
https://grasswiki.osgeo.org/wiki/GRASS_GSoC_Ideas_2024#STAC_(SpatioTemporal_Asset_Catalog)_Integration) and there are some overlaps with JSON idea, so I wonder if you would be interested in the STAC idea instead? 
If yes, for a test of skills please try to work on #3537.

**Comment by Shaf3eii** [comment]
I’m interested in contributing to this problem as part of GSoC 2025 and would love to work on this problem statement. Could you please confirm if this issue is still open for contribution?

**Comment by Majd725** [comment]
Hi @petrasovaa, I'm also interested in contributing to this problem for GSoC 2025. Are you still accepting solutions for this problem or is #5233 enough?

**Comment by Majd725** [comment]
Yes please. I'm interested in anything as I'm still a freshman just learning. I'm open to learning and contributing even if it's unpaid since I just want to learn.

**Comment by Majd725** [comment]
> > Hi [@petrasovaa](https://github.com/petrasovaa), I'm also interested in contributing to this problem for GSoC 2025. Are you still accepting solutions for this problem or is [#5233](https://github.com/OSGeo/grass/pull/5233) enough?
> 
> This is closed, but I could find something else if you know which topic you are interested in. I must say, it's getting rather late with the applications closing soon.

Yes please. I'm interested in anything as I'm still a freshman just learning. I'm open to learning and contributing even if it's unpaid since I just want to learn.

**Comment by petrasovaa** [comment]
> > > Hi [@petrasovaa](https://github.com/petrasovaa), I'm also interested in contributing to this problem for GSoC 2025. Are you still accepting solutions for this problem or is [#5233](https://github.com/OSGeo/grass/pull/5233) enough?
> > 
> > 
> > This is closed, but I could find something else if you know which topic you are interested in. I must say, it's getting rather late with the applications closing soon.
> 
> Yes please. I'm interested in anything as I'm still a freshman just learning. I'm open to learning and contributing even if it's unpaid since I just want to learn.

You can look at some [beginner issues](https://github.com/OSGeo/grass/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20first%20issue%22), e.g. #5442

Or you can write a test for a tool without tests, just check open PRs that somebody is not already working on it.

**Comment by mohamed-walid-fathy** [comment]
hello, I found about your project through GSoC, Can I work on the project even if I did not send a formal proposal?

**Comment by petrasovaa** [comment]
> hello, I found out about your project through GSoC, Can I work on the project even if I did not send a formal proposal? I am looking for a beginner-level bug/feature as this is my first time contributing to open source.

Sure, check out [issues for beginners](https://github.com/OSGeo/grass/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20first%20issue%22) and [contributing guidelines](https://github.com/OSGeo/grass/blob/main/CONTRIBUTING.md).

**Comment by ayushjariyal** [comment]
@petrasovaa, I am planning to apply for GSoC 2026 and found this issue via one of your project descriptions. I have prior experience contributing to other open-source projects and would appreciate your guidance on how to start contributing here.
